src/command.rs

```diff
@@ -49,6 +51,95 @@ macro_rules! delegate {
     }};
 }
 
+/// If a command is `OverSsh` then it can be executed over an SSH session.
+/// Primarily a way to allow `std::process::Command` to be turned directly into an `openssh::Command`.
+pub trait OverSsh {
+    /// Given a session, return a command that can be executed over that session.
+    ///
+    /// **Note**: The command to be executed on the remote machine does not include
+    /// any environment variables or the current directory. They must be
+    /// set explicitly, if desired.
```

Comment:
What does "set explicitly" mean here?

Reply:
Also, I wonder if there's an argument for us simply returning an error if we find that the current working directory or an environment variable is overridden?

Alternatively, can we transparently handle that as well? I guess probably not without assuming the remote end is Bash?

Reply:
The mux protocol actually supports env specification, though I didn't expose that because openssh didn't have that method.

For setting current dir, AFAIK, ssh does not support that.

Reply:
In that case, we should error (if possible) if `Command::current_dir` or `Command::env` has been called on the `Command` we're passed in.